File: java/binary search.java
import java.util.*;
class binarysearch
{

	public static void main(String[] args)
	{
		Scanner sc = new Scanner(System.in);
		binarysearch bs = new binarysearch();
		System.out.print("Enter the size:");
		int n=sc.nextInt();
		int[] A = new int[n];
		System.out.print("Enter the elements:");
		for(int i=0;i<n;i++)
				A[i] = sc.nextInt();
		System.out.print("Enter the item to be searched:");
		int item = sc.nextInt();	
		int k = bs.Binaryser(A,n,item);
		if(k!=0)
			System.out.println(item+" is found at position "+k);	
		else
			System.out.println("Element "+item+" not in the list");

	}

	int Binaryser(int A[], int n, int item)
	{
		int beg=0,end=n-1,mid;
		while(beg<=end)
		{
			mid=(beg+end)/2;
			if(A[mid]==item)
			{
				return(mid+1);
			}
			else if(item>A[mid])
			{
				beg=mid+1;
			}
			else
			{
				end=mid-1;
			}
		}
		return 0;
	}

}

File: cpp/Binarysearch.cpp
#include <iostream>
using namespace std;

int main()
{
    int Arr[15] = {6,8,13,17,20,22,25,28,30,35}, low = 0 , high = 9 , mid , srch;
    cout << "Enter the Element to search : ";
    cin >> srch;

    while(low <= high)
    {
        mid = (low + high)/2;
        if(srch == Arr[mid])
        {
            cout << "Search successful." << endl;
            cout << "The Element is found at " << mid << endl;
            return 0;
        }
        else if(srch < Arr[mid])
        {
            high = mid - 1;
        }
        else
        {
            low = mid + 1;
        }
    }
    cout << "Search unsuccessful." << endl;
    cout << "Element not found." << endl;


}
